### DISCUSSION TITLE
add kcm support for lxqt-confg

### DISCUSSION THREAD

**Discussion by antis81** [lxqt/lxqt#2064] [Ideas]
Our issues with lxqt-config vs KDE systemsettings is that those are concurring concepts:

- LXQt -> *.desktop files (docs??)
- KDE -> kcm binary plugins (https://techbase.kde.org/Development/Tutorials/KCM_HowTo)

This leads to a lot of redundancy and responsibility conflicts. For example take the "appearance" tool:

![lxqt_vs_appearance](https://user-images.githubusercontent.com/440517/131469676-8f61f343-742b-4a8e-be72-63225f5a2c76.png)

Sorry for it being in German (note to self: prepare a VM with English setup).

Adding kcm support to lxqt-config makes it both - light on resources and fully functional. This would:

a) introduce a standard (yeah, right!)
b) make life easier for packagers (kcm's in LXQt **and** KDE)
    -> create a `lxqt-systemsettings`
c) reduce need of writing new config tools over and over

And the best part:
Since we are using similar licensing models we can actually reuse existing code! The integration should be relatively simple and result in having the "best of both worlds". The relevant parts are in [kcmutils](https://github.com/KDE/kcmutils).

(Gnome is a different game though. I don't have experience with that. Anyone?)

**Comment by tsujan** [comment]
There may be a big misunderstanding here. `systemsettings` has no relation to LXQt. It's a QML monstrosity KDE devs have made for KDE. We never use kcm.

**Reply by tsujan** [reply]
In addition, KDE ways of doing things are by no means our standards.

**Reply by antis81** [reply]
You repeat the problem I am addressing! I think about solving it by implementing a minimal KCM bridge completely independent from KDE, but to integrate the modules we need.

Basically this is:
- KCM for kwin (don't know the name) 
- kcm_keys
- KCM for kwin compositor (optional)

~~Do we want to provide our configuration tool for every window manager we support? In that case we need to make a "list of supported WM's". Currently this list is empty and actually the only way to properly configure WM's, shortcuts and compositors is by having users learn their configuration files and edit them with a text editor. Do we actually want that?~~

**Reply by tsujan** [reply]
> Do we want to provide our configuration tool for every window manager we support?

We provide no config tool for any WM. obconf-qt was an exception but we removed it from 0.17.

> Currently this list is empty

See `/usr/share/lxqt/windowmanagers.conf`, although more can be used.

**Reply by antis81** [reply]
As @steff has evaluated it already we do not need an additional KCM bridge. It works through *.desktop files in [lxqt-kcm-integration](https://github.com/lxqt/lxqt-kcm-integration). So we can close this discussion.

**Reply by tsujan** [reply]
> So we can close this discussion.

Unlike issues, discussions have no close state and, as @stefonarch mentioned elsewhere, it's better not to delete them because they might be useful with Internet searches later.

**Reply by stefonarch** [reply]
> See `/usr/share/lxqt/windowmanagers.conf`, although more can be used.

Didn't know about that list,  it is used for autodetection in session-settings I suppose?

**Reply by tsujan** [reply]
> it is used for autodetection in session-settings I suppose

Yes. But, for example, you could easily use KWinFT with LXQt, although it isn't listed.

**Reply by tsujan** [reply]
And I can't help saying again and again how much I like this degree of modularity in LXQt ;)

**Reply by stefonarch** [reply]
I like it too, but on the other side some users  argue that it never feels like one piece, like an assembling.
I think our goal should be satisfying both needs as far as possible.

**Reply by antis81** [reply]
As @tsujan mentioned  it:
I use KWinFT - like butter! (Attention: Manjaro is not Debian! :smile:)

FYI:
Only adjustment that I had to make manually was the KDE task switcher (the `Breeze/-dark/-dusk` ones don't work!)
I use `MediumDefault` (works **only with KWinFT!**).
Since MediumDefault has a similar look to good old openbox task switcher, it doesn't "feel" too much different!

**Reply by tsujan** [reply]
> I use KWinFT - like butter!

I'll try it again if I find some time. The last time I tried it, it had shaky animations; otherwise, it inherited all properties and settings of KWin.

**Reply by tsujan** [reply]
> some users argue that it never feels like one piece, like an assembling.

Sadly, it may be seen so, only because of WM.

For obvious reasons, well-known DEs like XFCE and Cinnamon don't ruin what they've achieved by taking Wayland too seriously. Perhaps, we need a similar approach in the case of WM.

**Reply by tsujan** [reply]
I had a little time to try KWinFT again. After all, its installation is very easy and nothing needs to be changed after it.

I mostly use translucent windows with blurring. KWinFT's animations became choppy when more than 2 windows were shown, while KWin's are always smooth.

Window shading didn't work with KWinFT.

**Comment by antis81** [comment]
>I mostly use translucent windows with blurring. KWinFT's animations became choppy when more than 2 windows were shown, while KWin's are always smooth.

Didn't test blurring with KWinFT (I don't need/want it anyway). I run 2 hardware setups:
1. "refurbished" (and now pretty cool :sunglasses:) 15+ year old MSI with 3.5GHz dual-core CPU, 4GB RAM and nvidia 8500GT graphics
    - dual-monitor setup (1080p DVI and 1600x900(?) VGA)
    - (note: openbox worked fine with this setup and only minor issues ("wrong" selection order turned DVI screen "black"). With kwin(ft) I didn't try -> likely the same using `lxqt-config-monitor` tool!
1. Laptop 4GHz quad-core; 16 GiB RAM + intel SoC graphics

Both setups don't show performance issues (window shadows are :ok:) and show similar footprint ~400-500MB. This also depends on Manjaro services -> e.g. disabling "check for updates" in pamac seems to save ~100MB).

Both KWin & KWinFT have issues with window resizing:
When a window "sticks" to one screen side, I cannot "widen" it via mouse drag gesture. Can be annoying at times (workaround is possible by moving the window and drag on oposite edge).

**Reply by tsujan** [reply]
For me, KWin and KwinFT's memory usages are the same: between 35 and 50 MiB. I don't use Manjaro's automation (didn't like pamac and uninstalled it; only use pacman in terminal).

**Reply by antis81** [reply]
Talking about complete LXQt + kwin(ft) RAM usage btw. (not much different to openbox).

**Reply by tsujan** [reply]
> Talking about complete LXQt + kwin(ft) RAM usage

Yes, I know. KWin's resource usage is optimized and KWinFT inherits it.
